packages/framework/src/Command/DomainInfoCommand.php

```diff
@@ -0,0 +1,149 @@
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyInfo\Extractor\ReflectionExtractor;
+
+class DomainInfoCommand extends Command
+{
+    protected const ARG_PROPERTY_NAME = 'propertyName';
+    protected const ARG_ID = 'username';
+
+    protected const OPTION_ALL = 'all';
+
+    protected const RETURN_CODE_OK = 0;
+    protected const RETURN_CODE_ERROR = 1;
+
+    /**
+     * @var string
+     */
+    protected static $defaultName = 'shopsys:domains:info';
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    protected $domain;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(Domain $domain)
+    {
+        parent::__construct();
+
+        $this->domain = $domain;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure(): void
+    {
+        $this
+            ->setDescription('Loads and displays domain info.')
+            ->addArgument(static::ARG_PROPERTY_NAME, InputArgument::OPTIONAL, 'Property that should be loaded', 'id')
+            ->addArgument(static::ARG_ID, InputArgument::OPTIONAL, 'Domain ID (if omitted, the command will output all distinct values)')
+            ->addOption(static::OPTION_ALL, 'a', InputOption::VALUE_NONE, 'Display all property values (without sorting and deduplication)');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        try {
+            $domainConfigs = $this->getDomainConfigs($input);
+        } catch (\InvalidArgumentException $e) {
+            $io->error($e->getMessage());
+
+            return static::RETURN_CODE_ERROR;
+        }
+
+        $propertyAccessor = PropertyAccess::createPropertyAccessor();
+        $propertyName = $input->getArgument(static::ARG_PROPERTY_NAME);
+        $propertyValues = [];
+        foreach ($domainConfigs as $domainConfig) {
+            if (!$propertyAccessor->isReadable($domainConfig, $propertyName)) {
+                $this->outputPropertyNotAccessible($io, $domainConfig, $propertyName);
+
+                return static::RETURN_CODE_ERROR;
+            }
+
+            $propertyValues[] = $propertyAccessor->getValue($domainConfig, $propertyName);
+        }
+
+        if (!$input->getOption(static::OPTION_ALL)) {
+            sort($propertyValues);
+            $propertyValues = array_unique($propertyValues);
+        }
+        $io->writeln(implode("\t", $this->formatPropertyValues($propertyValues)));
```

Comment:
Wouldn't be better to print each value on a separate line? It's then easier to use this command in shell scripts like (without need to use regular expressions / awk, ... )
```
for x in `./bin/console shopsys:domains:info`; do ..... 
```

and to me personally, it's more understandable separated with new lines, then tabs

Reply:
:+1: Good idea. However, because of how Phing works with the output I left the possibility of tab-seaprated output using `--oneline` option. See 27578e7bf